crates/ide/src/references.rs

```diff
@@ -99,29 +100,40 @@ pub(crate) fn find_def(
     syntax: &SyntaxNode,
     offset: TextSize,
 ) -> Option<Definition> {
-    let def = match sema.find_node_at_offset_with_descend(syntax, offset)? {
-        ast::NameLike::NameRef(name_ref) => match NameRefClass::classify(sema, &name_ref)? {
-            NameRefClass::Definition(def) => def,
-            NameRefClass::FieldShorthand { local_ref, field_ref: _ } => {
-                Definition::Local(local_ref)
-            }
-        },
-        ast::NameLike::Name(name) => match NameClass::classify(sema, &name)? {
-            NameClass::Definition(it) | NameClass::ConstReference(it) => it,
-            NameClass::PatFieldShorthand { local_def, field_ref: _ } => {
-                Definition::Local(local_def)
-            }
-        },
-        ast::NameLike::Lifetime(lifetime) => NameRefClass::classify_lifetime(sema, &lifetime)
-            .and_then(|class| match class {
-                NameRefClass::Definition(it) => Some(it),
-                _ => None,
+    // FIXME: our macro handling here could be improved, ideally we'd disambiguate if the token maps
+    // to multiple defs, but LSP doesn't really allow that.
```

Comment:
We don’t care about LSP at this layer, we can (and should) return whatever makes most sense, and dumb it down in to_proto if necessary.

Reply:
Ye we should adjust this to handle multiple defs. I also think we should show the references for all definitions a token refers to if a macro uses the span for multiple defs when converting to proto.